src/directory/mmap_directory.rs

```diff
@@ -470,6 +441,30 @@ impl Directory for MmapDirectory {
     fn watch(&self, watch_callback: WatchCallback) -> crate::Result<WatchHandle> {
         Ok(self.inner.watch(watch_callback))
     }
+
+    fn sync_directory(&self) -> Result<(), io::Error> {
+        let mut open_opts = OpenOptions::new();
+
+        // Linux needs read to be set, otherwise returns EINVAL
+        // write must not be set, or it fails with EISDIR
+        open_opts.read(true);
```

Comment:
this code is also active in windows

Reply:
Yeah I genuinely have no clue if it has any effect what it does on windows... but it has the merit of having been tested by some user a long time ago :D

Reply:
I think it's additive, but if it has been tested should be fine :)